tests/datasets/public_dataset_test.py

```diff
@@ -78,3 +80,85 @@ class CustomPublicDataset(PublicDataset):
     assert dataset.root == expected_paths['root']
     assert dataset.path == expected_paths['path']
     assert dataset.downloads_rootpath == expected_paths['download']
+
+
+def test_dataset_download_fails(tmp_path):
+    class CustomPublicDataset(PublicDataset):
```

Comment:
can you use this as a fixture for all the tests?
two mirrors should be sufficient for all cases, but they must be different.